auth_oidc/models/auth_oauth_provider.py

```diff
@@ -68,15 +66,21 @@ def _parse_id_token(self, id_token, access_token):
         self.ensure_one()
         res = {}
         header = jwt.get_unverified_header(id_token)
-        res.update(
-            jwt.decode(
-                id_token,
-                self._get_key(header.get("kid")),
-                algorithms=["RS256"],
-                audience=self.client_id,
-                access_token=access_token,
-            )
-        )
-
+        error = None
+        for key in self._get_keys(header.get("kid")):
```

Comment:
i interpreted the following:
>  If there are multiple keys in the referenced JWK Set document, a kid value MUST be provided in the JOSE Header
from [OpenID Connect Core Section 10.1](https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.10.1)

As the `kid` is optional. And missed the as long as there is only one key in the JWK Set (jwks_uri) part. So with your question in mind, it should be unique and not a list. (An our current provider implementation is wrong). I will update the merge request.

Reply:
i updated the commit to only handle cases for JWKS with a single key

Reply:
Well, as i started checking our implementation of JWK, i checked the docs again

[OpenID ID Token Validation](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation)
Would be our starting point.
Step 6:
> If the ID Token is received via direct communication between the Client and the Token Endpoint (which it is in this flow), the TLS server validation MAY be used to validate the issuer in place of checking the token signature. The Client MUST validate the signature of all other ID Tokens according to [JWS](https://openid.net/specs/openid-connect-core-1_0.html#JWS) [JWS] using the algorithm specified in the JWT alg Header Parameter. The Client MUST use the keys provided by the Issuer.

Would be the one that defines we MUST validate the signature according to JWS

[JWS Note on Key Seelection](https://datatracker.ietf.org/doc/html/draft-ietf-jose-json-web-signature#appendix-D)
Defines which keys should be considered.
Step 2:
>Filter the set of collected keys.  For instance, some applications will use only keys referenced by "kid" (key ID) or "x5t" (X.509 certificate SHA-1 thumbprint) parameters. If the application uses the "alg" (algorithm), "use" (public key use), or "key_ops" (key operations) parameters, keys with keys with inappropriate values of those parameters would be excluded.       Additionally, keys might be filtered to include or exclude keys with certain other member values in an application specific manner.  For some applications, no filtering will be applied

And the following sections all sound like we should allow multiple keys (so the initial for loop was more correct than the current implementation)

So it seems for OpenID Connect:
* when no kid is specified in ID Token, the JWKS can only contain one key (According to the OpenID Connect Spec)
* when kid is specified in ID Token we need to find the matching Key from JWKS because there can be multiple keys for the same kid. The note at the end of 
[JWS Note on Key Seelection](https://datatracker.ietf.org/doc/html/draft-ietf-jose-json-web-signature#appendix-D) explicitly allow the behaviour of checking the signature against all keys with the same kid:
> Note that it is reasonable for some applications to perform signature
   or MAC validation prior to making a trust decision about a key, since
   keys for which the validation fails need no trust decision.

As JWS only specifies key selection i would conclude that if ID Token gives no kid it is just a missing hint for the correct key so it should still be allowed to have an ID Token without a kid and a JWKS with a single key that has a kid and validate against it.